### DISCUSSION TITLE
How to make aiogram 3 filter to enable in handler only one command or without any command execution

### DISCUSSION THREAD

**Discussion by RomanVD39** [aiogram/aiogram#1429] [General]
Need to make a dispatcher filter to execute a code only if message.text has inside a specified command or doesn't have any command.

I have tried this and many others combinations of aiogram 3 F:

`@dp.message(F.text and (~(F.entities[...].type == MessageEntityType.BOT_COMMAND) or F.text.startswith("/mycommand")))`

but it doesn't work properly.

Any suggestions? How to mix F filter parameters correctly?

**Comment by Olegt0rr** [comment]
Try this:

```python
@dp.message(Command("mycommand"))
@dp.message(F.text & ~F.text.startswith('/'))
async def handler(...): ...

```

P.S.: You didn't provide any examples and I don't have enough time to test it properly.
P.P.S: Combining of F-filters is [described in the docs](https://docs.aiogram.dev/en/dev-3.x/dispatcher/filters/magic_filters.html#combining)

**Reply by RomanVD39** [reply]
Great thanks, it works fine.

But what do you think about
`~(F.entities[...].type == MessageEntityType.BOT_COMMAND)`
is it the same for
`~F.text.startswith('/')`

PS Here is the information about how the filters work. These are not just pure logical operations when combined. Sequencing is a key. At the end of the article some pictures:
https://docs.aiogram.dev/en/latest/dispatcher/router.html
